lib/rubocop/ast/processed_source.rb

```diff
@@ -240,6 +272,13 @@ def create_parser(ruby_version)
           end
         end
       end
+
+      def sort_tokens_if_needed
+        return if @tokens_sorted
+
+        tokens.sort_by!(&:begin_pos)
```

Comment:
Tokens should be frozen. I think it's not the best idea to mutate it anyways; best to have a method `sorted_tokens` I think. FYI, it doesn't require more memory allocation, only both arrays won't be garbage collectable while we process the file instead of one.